lib/taskana-core/src/main/java/pro/taskana/task/internal/models/TaskSummaryImpl.java

```diff
@@ -776,6 +778,78 @@ public void setCustomInt8(Integer customInt8) {
     this.customInt8 = customInt8;
   }
 
+  @Override
+  public Task asTask() {
```

Comment:
Why do we need this?

Reply:
It is used to convert tasksummaries obrained from the query to tasks, which will be passed to the function taskanaEngine.getTaskRoutingManager().determineWorkbasketId(task). determineWorkbasketId() only accept task as parameter, not tasksummary instances.

Casting cannot be used here, this is the error thrown when trying to cast:
Caused by: java.lang.ClassCastException: Cannot cast pro.taskana.task.internal.models.TaskSummaryImpl to pro.taskana.task.api.models.Task